src/table/__tests__/expandable-rows.test.tsx

```diff
@@ -0,0 +1,272 @@
+    expect(table.findExpandToggle(3)!.getElement()).toHaveAttribute('aria-expanded', 'false');
+  });
+
+  test('expand toggles labels are set with i18n provider', () => {
+    const { table } = renderTable({
+      items: simpleItems,
+      expandableRows: {
+        isItemExpandable: item => item.expandable,
+        expandedItems: [simpleItems[0]],
+        getItemChildren: () => [],
+        onExpandableItemToggle: () => {},
+      },
+      messages: {
+        'ariaLabels.expandButtonLabel': 'Expand item',
+        'ariaLabels.collapseButtonLabel': 'Collapse item',
+      },
+    });
+    expect(table.findExpandToggle(1)!.getElement()).toHaveAccessibleName('Collapse item');
+    expect(table.findExpandToggle(1)!.getElement()).toHaveAttribute('aria-expanded', 'true');
+    expect(table.findExpandToggle(3)!.getElement()).toHaveAccessibleName('Expand item');
+    expect(table.findExpandToggle(3)!.getElement()).toHaveAttribute('aria-expanded', 'false');
+  });
+
+  test('nested items are rendered for expandable and expanded items', () => {
+    const { table } = renderTable({
+      items: nestedItems,
+      expandableRows: {
+        isItemExpandable: item => item.children && item.children.length > 0,
+        expandedItems: flatten(nestedItems).filter(item => item.name !== 'Root-2'),
+        getItemChildren: item => item.children ?? [],
+        onExpandableItemToggle: () => {},
+      },
+    });
+    expect(table.findRows().map(r => r.find('td')!.getElement().textContent)).toEqual([
+      'Root-1',
+      'Nested-1.1',
+      'Nested-1.2',
+      'Nested-1.3',
+      'Nested-1.3.1',
+      'Nested-1.3.2',
+      'Root-2',
+    ]);
+    // Root-1
+    expect(table.findRows()[0].getElement()).toHaveAttribute('aria-level', '1');
+    expect(table.findRows()[0].getElement()).toHaveAttribute('aria-setsize', '2');
+    expect(table.findRows()[0].getElement()).toHaveAttribute('aria-posinset', '1');
+    // Nested-1.3.2
+    expect(table.findRows()[5].getElement()).toHaveAttribute('aria-level', '3');
+    expect(table.findRows()[5].getElement()).toHaveAttribute('aria-setsize', '2');
+    expect(table.findRows()[5].getElement()).toHaveAttribute('aria-posinset', '2');
+  });
+
+  test('onExpandableItemToggle fires with item and expand state', () => {
+    const onExpandableItemToggle = jest.fn();
+    const { table } = renderTable({
+      items: simpleItems,
+      expandableRows: {
+        isItemExpandable: item => item.expandable,
+        expandedItems: [simpleItems[0]],
+        getItemChildren: () => [],
+        onExpandableItemToggle,
+      },
+    });
+
+    table.findExpandToggle(1)!.click();
+    expect(onExpandableItemToggle).toBeCalledTimes(1);
+    expect(onExpandableItemToggle).toBeCalledWith(
+      expect.objectContaining({ detail: { item: simpleItems[0], expanded: false } })
+    );
+
+    table.findExpandToggle(3)!.click();
+    expect(onExpandableItemToggle).toBeCalledTimes(2);
```

Comment:
Why is it called twice while expanding? Or is it because the function was called before this as well?

Reply:
The event counter is cumulative. For each `.click()` issued the asserted count needs to be increased.